lib/rack/mock.rb

```diff
@@ -212,8 +212,15 @@ def body
       #   end
       buffer = String.new
 
-      super.each do |chunk|
-        buffer << chunk
+      body = super
+      body = body.body if body.respond_to?(:body)
+
+      if body.respond_to?(:each)
+         body.each do |chunk|
+            buffer << chunk
+         end
+      else
+         buffer = body.to_s
```

Comment:
There's an extra space of indentation here.

```suggestion
      if body.respond_to?(:each)
        body.each do |chunk|
          buffer << chunk
        end
      else
        buffer = body.to_s
```